Include/pymath.h

```diff
@@ -56,24 +56,8 @@
  * doesn't support NaNs.
  */
 #if !defined(Py_NAN) && !defined(Py_NO_NAN)
-#  if !defined(__INTEL_COMPILER)
-#    define Py_NAN (Py_HUGE_VAL * 0.)
-#  else /* __INTEL_COMPILER */
-#    if defined(ICC_NAN_STRICT)
-        #pragma float_control(push)
-        #pragma float_control(precise, on)
-        #pragma float_control(except,  on)
-        Py_NO_INLINE static double __icc_nan()
-        {
-            return sqrt(-1.0);
-        }
-        #pragma float_control (pop)
-#       define Py_NAN __icc_nan()
-#    else /* ICC_NAN_RELAXED as default for Intel Compiler */
-        static const union { unsigned char buf[8]; double __icc_nan; } __nan_store = {0,0,0,0,0,0,0xf8,0x7f};
-#       define Py_NAN (__nan_store.__icc_nan)
-#    endif /* ICC_NAN_STRICT */
-#  endif /* __INTEL_COMPILER */
+   // Use C99 "NAN" constant: quiet Not-A-Number (when supported)
+#  define Py_NAN NAN
```

Comment:
Here you're changing the type of the `Py_NAN` constant from `double` to `float`. That _might_ be benign, or it might not. Either way, it's probably worth using `((double)NAN)` instead of plain `NAN` as the macro value.

Reply:
Would it make sense to declare a Py_NAN constant? Something like: ``PyAPI_DATA(double) Py_NAN;``.

The macro NAN can require computing the value or call a function. I don't think that <math.h> ensures that it's directly a value, even if it's the case if most cases.

Reply:
> Would it make sense to declare a Py_NAN constant?

If we can make that work, then sure. How would the constant value be assigned?

Reply:
> It's probably worth using ((double)NAN) instead

Oh, I didn't notice that it's a float! Fixed.